src/renderer/src/App.tsx

```diff
@@ -26,16 +27,26 @@ const PersistedProjectIdStore = createActiveProjectIdStore({
 })
 
 export const App = () => (
-	<ThemeProvider theme={theme}>
-		<CssBaseline />
-		<IntlProvider>
-			<QueryClientProvider client={queryClient}>
+	<ReuseableProviderWrapper>
+		<QueryClientProvider client={queryClient}>
+			<ApiProvider>
 				<ActiveProjectIdProvider store={PersistedProjectIdStore}>
-					<ApiProvider>
-						<RouterProvider router={router} />
-					</ApiProvider>
+					<RouterProvider router={router} />
 				</ActiveProjectIdProvider>
-			</QueryClientProvider>
-		</IntlProvider>
-	</ThemeProvider>
+			</ApiProvider>
+		</QueryClientProvider>
+	</ReuseableProviderWrapper>
 )
+
+export const ReuseableProviderWrapper = ({
+	children,
+}: {
+	children: ReactNode
+}) => {
+	return (
+		<ThemeProvider theme={theme}>
+			<CssBaseline />
+			<IntlProvider>{children}</IntlProvider>
+		</ThemeProvider>
+	)
+}
```

Comment:
i don't think abstracting this is necessary for the time being. would suggest just doing duplicate work of setting up the providers in the relevant test helper for now